Reduce header includes in Node.h, StyledElement.h, and HTMLElement.h

rniwa opened the pull request:
#### ea9b0da5a95024c7c613a93baca2d9218196130d
<pre>
Reduce header includes in Node.h, StyledElement.h, and HTMLElement.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=292316">https://bugs.webkit.org/show_bug.cgi?id=292316</a>

Reviewed by Chris Dumez.

Move more functions to *Inlines.h to reduce header includes in Node.h, StyledElement.h, and HTMLElement.h.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
* Source/WebCore/SaferCPPExpectations/ForwardDeclCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXTextMarker.cpp:
* Source/WebCore/accessibility/AccessibilityListBoxOption.cpp:
* Source/WebCore/accessibility/atspi/AccessibilityObjectImageAtspi.cpp:
* Source/WebCore/accessibility/cocoa/AXCoreObjectCocoa.mm:
* Source/WebCore/accessibility/mac/AccessibilityObjectMac.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
* Source/WebCore/animation/KeyframeEffect.cpp:
* Source/WebCore/bindings/js/JSEventListener.h:
* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/CSSStyleProperties.h:
* Source/WebCore/css/IsImportant.h: Added.
* Source/WebCore/dom/BoundaryPoint.cpp:
(WebCore::BoundaryPoint::protectedDocument const): Moved to BoundaryPointInlines.h.
* Source/WebCore/dom/BoundaryPoint.h:
(WebCore::BoundaryPoint::document const): Ditto.
* Source/WebCore/dom/BoundaryPointInlines.h:
(WebCore::BoundaryPoint::document const):
(WebCore::BoundaryPoint::protectedDocument const):
* Source/WebCore/dom/ContainerNode.cpp:
(WebCore::ContainerNode::absoluteEventHandlerBounds):
* Source/WebCore/dom/ContainerNode.h:
(WebCore::ContainerNode::protectedRootNode const): Moved to TreeScopeInlines.h
(WebCore::ContainerNode::absoluteEventHandlerBounds): Moved to ContainerNode.cpp.
(WebCore::TreeScope::rootNode const): Moved to TreeScopeInlines.h
(WebCore::ContainerNode::rootNode const): Ditto.
* Source/WebCore/dom/DataTransferMac.mm:
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/DocumentMarkerController.cpp:
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementIteratorAssertions.h:
* Source/WebCore/dom/Node.cpp:
* Source/WebCore/dom/Node.h:
(WebCore::Node::pixelSnappedAbsoluteBoundingRect): Moved to NodeInlines.h
(WebCore::Node::document const): Ditto.
(WebCore::Node::protectedTreeScope const): Ditto.
* Source/WebCore/dom/NodeInlines.h:
(WebCore::Node::document const):
(WebCore::Node::protectedTreeScope const):
(WebCore::Node::pixelSnappedAbsoluteBoundingRect):
* Source/WebCore/dom/Position.h:
* Source/WebCore/dom/Range.cpp:
* Source/WebCore/dom/RenderedDocumentMarker.h:
* Source/WebCore/dom/ShadowRoot.h:
* Source/WebCore/dom/StyledElement.h:
* Source/WebCore/dom/TreeScope.h:
* Source/WebCore/dom/TreeScopeInlines.h:
(WebCore::TreeScope::rootNode const):
(WebCore::ContainerNode::rootNode const):
(WebCore::ContainerNode::protectedRootNode const):
* Source/WebCore/dom/ViewTransition.cpp:
* Source/WebCore/editing/AlternativeTextController.cpp:
* Source/WebCore/editing/ReplaceRangeWithTextCommand.cpp:
* Source/WebCore/editing/SelectionGeometryGatherer.cpp:
* Source/WebCore/editing/SpellingCorrectionCommand.cpp:
* Source/WebCore/editing/TextIterator.cpp:
* Source/WebCore/editing/cocoa/DataDetection.mm:
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
* Source/WebCore/editing/markup.cpp:
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::setPopover):
* Source/WebCore/html/HTMLElement.h:
(WebCore::HTMLElement::setPopover): Deleted.
* Source/WebCore/html/MediaElementSession.cpp:
* Source/WebCore/html/track/TextTrackCue.cpp:
(WebCore::TextTrackCue::protectedDocument const):
* Source/WebCore/html/track/TextTrackCue.h:
(WebCore::TextTrackCue::protectedDocument const): Moved to TextTrackCue.cpp.
* Source/WebCore/html/track/VTTCue.h:
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
* Source/WebCore/inspector/agents/page/PageTimelineAgent.cpp:
* Source/WebCore/layout/layouttree/LayoutTreeBuilder.cpp:
* Source/WebCore/loader/ImageLoader.h:
* Source/WebCore/loader/archive/cf/LegacyWebArchive.cpp:
* Source/WebCore/page/AutoscrollController.cpp:
* Source/WebCore/page/DragController.cpp:
* Source/WebCore/page/Page.cpp:
* Source/WebCore/page/ios/EventHandlerIOS.mm:
* Source/WebCore/page/mac/ServicesOverlayController.mm:
* Source/WebCore/page/scrolling/ScrollingCoordinator.cpp:
* Source/WebCore/page/text-extraction/TextExtraction.cpp:
* Source/WebCore/platform/cocoa/DragImageCocoa.mm:
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.mm:
* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
* Source/WebCore/rendering/AttachmentLayout.mm:
* Source/WebCore/rendering/RenderBox.h:
(WebCore::RenderBox::topLeftLocation const): Moved to RenderBoxInlines.h.
(WebCore::RenderBox::topLeftLocationOffset const): Ditto.
* Source/WebCore/rendering/RenderBoxInlines.h:
(WebCore::RenderBox::topLeftLocation const):
(WebCore::RenderBox::topLeftLocationOffset const):
* Source/WebCore/rendering/RenderLayer.h:
* Source/WebCore/rendering/RenderLayerInlines.h:
(WebCore::RenderLayer::page const):
(WebCore::RenderLayer::protectedPage const):
* Source/WebCore/rendering/RenderObject.cpp:
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::isDocumentElementRenderer const): Moved to RenderObjectInlines.h.
(WebCore::RenderObject::view const): Ditto.
(WebCore::RenderObject::document const): Ditto.
(WebCore::RenderObject::protectedTreeScopeForSVGReferences const): Ditto.
(WebCore::RenderObject::protectedFrame const): Ditto.
(WebCore::RenderObject::settings const): Ditto.
(WebCore::RenderObject::frame const): Ditto.
(WebCore::RenderObject::page const): Ditto.
(WebCore::RenderObject::protectedPage const): Ditto.
(WebCore::RenderObject::renderTreeBeingDestroyed const): Ditto.
* Source/WebCore/rendering/RenderObjectInlines.h:
(WebCore::RenderObject::document const):
(WebCore::RenderObject::checkedStyle const):
(WebCore::RenderObject::protectedTreeScopeForSVGReferences const):
(WebCore::RenderObject::isDocumentElementRenderer const):
(WebCore::RenderObject::view const):
(WebCore::RenderObject::frame const):
(WebCore::RenderObject::protectedFrame const):
(WebCore::RenderObject::page const):
(WebCore::RenderObject::protectedPage const):
(WebCore::RenderObject::settings const):
(WebCore::RenderObject::renderTreeBeingDestroyed const):
* Source/WebCore/rendering/RenderTable.h:
(WebCore::RenderTable::createAnonymousBoxWithSameTypeAs const): Moved to RenderTableInlines.h.
* Source/WebCore/rendering/RenderTableCell.h:
(WebCore::RenderTableCell::createAnonymousBoxWithSameTypeAs const): Moved to RenderTableCellInlines.h.
* Source/WebCore/rendering/RenderTableCellInlines.h:
(WebCore::RenderTableCell::createAnonymousBoxWithSameTypeAs const):
* Source/WebCore/rendering/RenderTableInlines.h:
(WebCore::RenderTable::createAnonymousBoxWithSameTypeAs const):
* Source/WebCore/rendering/RenderTableSection.h:
(WebCore::RenderTableSection::createAnonymousBoxWithSameTypeAs const): Moved to RenderTableSectionInlines.h.
* Source/WebCore/rendering/RenderTableSectionInlines.h:
(WebCore::RenderTableSection::createAnonymousBoxWithSameTypeAs const):
* Source/WebCore/rendering/style/StyleFilterImage.cpp:
* Source/WebCore/rendering/style/StylePaintImage.cpp:
* Source/WebCore/rendering/svg/RenderSVGText.h:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGContainer.cpp:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGImage.cpp:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGResource.cpp:
* Source/WebCore/style/StyleResolveForDocument.cpp:
* Source/WebCore/svg/SVGAElement.cpp:
* Source/WebCore/svg/SVGTextContentElement.cpp:
* Source/WebCore/testing/Internals.cpp:
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMRange.mm:
* Source/WebKit/WebProcess/InjectedBundle/DOM/InjectedBundleRangeHandle.cpp:
* Source/WebKit/WebProcess/WebPage/FindController.cpp:
* Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
* Source/WebKitLegacy/mac/DOM/DOM.mm:
* Source/WebKitLegacy/mac/DOM/DOMRange.mm:
* Source/WebKitLegacy/mac/DOM/WebDOMOperations.mm:
* Source/WebKitLegacy/mac/WebCoreSupport/WebValidationMessageClient.mm:
* Source/WebKitLegacy/mac/WebView/WebFullScreenController.mm:
* Tools/TestWebKitAPI/Tests/WebCore/MarkedText.cpp:

Canonical link: <a href="https://commits.webkit.org/294370@main">https://commits.webkit.org/294370@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b44aad576651548bf913134a007381a5795e7687

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101586 "5 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21254 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11567 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106744 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52219 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/103626 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21562 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/29750 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77365 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34392 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104593 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16652 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91743 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57702 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16477 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/9760 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51567 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86345 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9837 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/109097 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28719 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/21137 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86340 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/29080 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/87944 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/85904 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30648 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8361 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/22856 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16530 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28647 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/33936 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28458 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31781 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/30017 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/css/IsImportant.h:
```diff
@@ -0,0 +1,28 @@
+/*
+ * (C) 1999-2003 Lars Knoll (knoll@kde.org)
+ * Copyright (C) 2004, 2005, 2006 Apple Inc.
+ * Copyright (C) 2013 Intel Corporation. All rights reserved.
```
Too much copypasta?

rniwa replied:
No, we need to retain the original copyright from `CSSProperty.h` because I'm just copying the code.

annevk commented on Source/WebCore/dom/TreeScopeInlines.h:
```diff
@@ -62,4 +67,17 @@ inline bool TreeScope::containsMultipleElementsWithName(const AtomString& name)
     return m_elementsByName && !name.isEmpty() && m_elementsByName->containsMultiple(name);
 }
 
+// FIXME: Move these to ContainerNodeInlines.h
```
Any reason not to do this straight away or is that why this is a draft?

rniwa replied:
I'm trying to limit the scope of this PR. It can grow out of control if I start enforcing that everywhere.

cdumez commented on Source/WebCore/dom/BoundaryPoint.h:
```diff
@@ -35,8 +35,8 @@ struct BoundaryPoint {
 
-    Document& document() const;
-    WEBCORE_EXPORT Ref<Document> protectedDocument() const;
+    inline Document& document() const;
```
Personally, I like adding a comment stating *where* the function is defined.

rniwa replied:
Okay. I felt a bit redundant when functions are marked inline and defined in the corresponding `Inlines.h` file but we can add a comment.

cdumez commented on Source/WebCore/dom/BoundaryPoint.h:
```diff
@@ -35,8 +35,8 @@ struct BoundaryPoint {
-    Document& document() const;
-    WEBCORE_EXPORT Ref<Document> protectedDocument() const;
+    inline Document& document() const;
+    inline Ref<Document> protectedDocument() const;
```
ditto.

rniwa replied:
Fixed.

cdumez commented on Source/WebCore/dom/Node.h:
```diff
@@ -402,7 +408,7 @@ class Node : public EventTarget, public CanMakeCheckedPtr<Node> {
 
     WEBCORE_EXPORT LayoutRect absoluteBoundingRect(bool* isReplaced);
-    IntRect pixelSnappedAbsoluteBoundingRect(bool* isReplaced) { return snappedIntRect(absoluteBoundingRect(isReplaced)); }
+    inline IntRect pixelSnappedAbsoluteBoundingRect(bool* isReplaced);
```
ditto.

rniwa replied:
Fixed.

cdumez commented on Source/WebCore/dom/Node.h:
```diff
@@ -411,15 +417,15 @@ class Node : public EventTarget, public CanMakeCheckedPtr<Node> {
 
     // Returns the document associated with this node. A document node returns itself.
-    Document& document() const { return treeScope().documentScope(); }
+    inline Document& document() const;
```
ditto.

rniwa replied:
Fixed.

cdumez commented on Source/WebCore/dom/Node.h:
```diff
@@ -411,15 +417,15 @@ class Node : public EventTarget, public CanMakeCheckedPtr<Node> {
         return *m_treeScope;
     }
-    Ref<TreeScope> protectedTreeScope() const { return treeScope(); }
+    inline Ref<TreeScope> protectedTreeScope() const;
```
ditto.

rniwa replied:
Fixed.

cdumez commented on Source/WebCore/dom/TreeScope.h:
```diff
@@ -132,8 +132,8 @@ class TreeScope : public NoVirtualDestructorBase {
 
-    inline ContainerNode& rootNode() const; // Defined in ContainerNode.h
-    Ref<ContainerNode> protectedRootNode() const;
+    inline ContainerNode& rootNode() const;
```
ditto.

rniwa replied:
Fixed.

cdumez commented on Source/WebCore/dom/TreeScope.h:
```diff
@@ -132,8 +132,8 @@ class TreeScope : public NoVirtualDestructorBase {
-    inline ContainerNode& rootNode() const; // Defined in ContainerNode.h
-    Ref<ContainerNode> protectedRootNode() const;
+    inline ContainerNode& rootNode() const;
+    inline Ref<ContainerNode> protectedRootNode() const;
```
ditto.

rniwa replied:
Fixed.

rniwa committed:
Reduce header includes in Node.h, StyledElement.h, and HTMLElement.h
https://bugs.webkit.org/show_bug.cgi?id=292316

Reviewed by Chris Dumez.

Move more functions to *Inlines.h to reduce header includes in Node.h, StyledElement.h, and HTMLElement.h.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
* Source/WebCore/SaferCPPExpectations/ForwardDeclCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AXTextMarker.cpp:
* Source/WebCore/accessibility/AccessibilityListBoxOption.cpp:
* Source/WebCore/accessibility/atspi/AccessibilityObjectImageAtspi.cpp:
* Source/WebCore/accessibility/cocoa/AXCoreObjectCocoa.mm:
* Source/WebCore/accessibility/mac/AccessibilityObjectMac.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
* Source/WebCore/animation/KeyframeEffect.cpp:
* Source/WebCore/bindings/js/JSEventListener.h:
* Source/WebCore/css/CSSProperty.h:
* Source/WebCore/css/CSSStyleProperties.h:
* Source/WebCore/css/IsImportant.h: Added.
* Source/WebCore/dom/BoundaryPoint.cpp:
(WebCore::BoundaryPoint::protectedDocument const): Moved to BoundaryPointInlines.h.
* Source/WebCore/dom/BoundaryPoint.h:
(WebCore::BoundaryPoint::document const): Ditto.
* Source/WebCore/dom/BoundaryPointInlines.h:
(WebCore::BoundaryPoint::document const):
(WebCore::BoundaryPoint::protectedDocument const):
* Source/WebCore/dom/ContainerNode.cpp:
(WebCore::ContainerNode::absoluteEventHandlerBounds):
* Source/WebCore/dom/ContainerNode.h:
(WebCore::ContainerNode::protectedRootNode const): Moved to TreeScopeInlines.h
(WebCore::ContainerNode::absoluteEventHandlerBounds): Moved to ContainerNode.cpp.
(WebCore::TreeScope::rootNode const): Moved to TreeScopeInlines.h
(WebCore::ContainerNode::rootNode const): Ditto.
* Source/WebCore/dom/DataTransferMac.mm:
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/DocumentMarkerController.cpp:
* Source/WebCore/dom/Element.h:
* Source/WebCore/dom/ElementIteratorAssertions.h:
* Source/WebCore/dom/Node.cpp:
* Source/WebCore/dom/Node.h:
(WebCore::Node::pixelSnappedAbsoluteBoundingRect): Moved to NodeInlines.h
(WebCore::Node::document const): Ditto.
(WebCore::Node::protectedTreeScope const): Ditto.
* Source/WebCore/dom/NodeInlines.h:
(WebCore::Node::document const):
(WebCore::Node::protectedTreeScope const):
(WebCore::Node::pixelSnappedAbsoluteBoundingRect):
* Source/WebCore/dom/Position.h:
* Source/WebCore/dom/Range.cpp:
* Source/WebCore/dom/RenderedDocumentMarker.h:
* Source/WebCore/dom/ShadowRoot.h:
* Source/WebCore/dom/StyledElement.h:
* Source/WebCore/dom/TreeScope.h:
* Source/WebCore/dom/TreeScopeInlines.h:
(WebCore::TreeScope::rootNode const):
(WebCore::ContainerNode::rootNode const):
(WebCore::ContainerNode::protectedRootNode const):
* Source/WebCore/dom/ViewTransition.cpp:
* Source/WebCore/editing/AlternativeTextController.cpp:
* Source/WebCore/editing/ReplaceRangeWithTextCommand.cpp:
* Source/WebCore/editing/SelectionGeometryGatherer.cpp:
* Source/WebCore/editing/SpellingCorrectionCommand.cpp:
* Source/WebCore/editing/TextIterator.cpp:
* Source/WebCore/editing/cocoa/DataDetection.mm:
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
* Source/WebCore/editing/markup.cpp:
* Source/WebCore/html/HTMLElement.cpp:
(WebCore::HTMLElement::setPopover):
* Source/WebCore/html/HTMLElement.h:
(WebCore::HTMLElement::setPopover): Deleted.
* Source/WebCore/html/MediaElementSession.cpp:
* Source/WebCore/html/track/TextTrackCue.cpp:
(WebCore::TextTrackCue::protectedDocument const):
* Source/WebCore/html/track/TextTrackCue.h:
(WebCore::TextTrackCue::protectedDocument const): Moved to TextTrackCue.cpp.
* Source/WebCore/html/track/VTTCue.h:
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
* Source/WebCore/inspector/agents/page/PageTimelineAgent.cpp:
* Source/WebCore/layout/layouttree/LayoutTreeBuilder.cpp:
* Source/WebCore/loader/ImageLoader.h:
* Source/WebCore/loader/archive/cf/LegacyWebArchive.cpp:
* Source/WebCore/page/AutoscrollController.cpp:
* Source/WebCore/page/DragController.cpp:
* Source/WebCore/page/Page.cpp:
* Source/WebCore/page/ios/EventHandlerIOS.mm:
* Source/WebCore/page/mac/ServicesOverlayController.mm:
* Source/WebCore/page/scrolling/ScrollingCoordinator.cpp:
* Source/WebCore/page/text-extraction/TextExtraction.cpp:
* Source/WebCore/platform/cocoa/DragImageCocoa.mm:
* Source/WebCore/platform/cocoa/PlaybackSessionModelMediaElement.mm:
* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
* Source/WebCore/rendering/AttachmentLayout.mm:
* Source/WebCore/rendering/RenderBox.h:
(WebCore::RenderBox::topLeftLocation const): Moved to RenderBoxInlines.h.
(WebCore::RenderBox::topLeftLocationOffset const): Ditto.
* Source/WebCore/rendering/RenderBoxInlines.h:
(WebCore::RenderBox::topLeftLocation const):
(WebCore::RenderBox::topLeftLocationOffset const):
* Source/WebCore/rendering/RenderLayer.h:
* Source/WebCore/rendering/RenderLayerInlines.h:
(WebCore::RenderLayer::page const):
(WebCore::RenderLayer::protectedPage const):
* Source/WebCore/rendering/RenderObject.cpp:
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::isDocumentElementRenderer const): Moved to RenderObjectInlines.h.
(WebCore::RenderObject::view const): Ditto.
(WebCore::RenderObject::document const): Ditto.
(WebCore::RenderObject::protectedTreeScopeForSVGReferences const): Ditto.
(WebCore::RenderObject::protectedFrame const): Ditto.
(WebCore::RenderObject::settings const): Ditto.
(WebCore::RenderObject::frame const): Ditto.
(WebCore::RenderObject::page const): Ditto.
(WebCore::RenderObject::protectedPage const): Ditto.
(WebCore::RenderObject::renderTreeBeingDestroyed const): Ditto.
* Source/WebCore/rendering/RenderObjectInlines.h:
(WebCore::RenderObject::document const):
(WebCore::RenderObject::checkedStyle const):
(WebCore::RenderObject::protectedTreeScopeForSVGReferences const):
(WebCore::RenderObject::isDocumentElementRenderer const):
(WebCore::RenderObject::view const):
(WebCore::RenderObject::frame const):
(WebCore::RenderObject::protectedFrame const):
(WebCore::RenderObject::page const):
(WebCore::RenderObject::protectedPage const):
(WebCore::RenderObject::settings const):
(WebCore::RenderObject::renderTreeBeingDestroyed const):
* Source/WebCore/rendering/RenderTable.h:
(WebCore::RenderTable::createAnonymousBoxWithSameTypeAs const): Moved to RenderTableInlines.h.
* Source/WebCore/rendering/RenderTableCell.h:
(WebCore::RenderTableCell::createAnonymousBoxWithSameTypeAs const): Moved to RenderTableCellInlines.h.
* Source/WebCore/rendering/RenderTableCellInlines.h:
(WebCore::RenderTableCell::createAnonymousBoxWithSameTypeAs const):
* Source/WebCore/rendering/RenderTableInlines.h:
(WebCore::RenderTable::createAnonymousBoxWithSameTypeAs const):
* Source/WebCore/rendering/RenderTableSection.h:
(WebCore::RenderTableSection::createAnonymousBoxWithSameTypeAs const): Moved to RenderTableSectionInlines.h.
* Source/WebCore/rendering/RenderTableSectionInlines.h:
(WebCore::RenderTableSection::createAnonymousBoxWithSameTypeAs const):
* Source/WebCore/rendering/style/StyleFilterImage.cpp:
* Source/WebCore/rendering/style/StylePaintImage.cpp:
* Source/WebCore/rendering/svg/RenderSVGText.h:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGContainer.cpp:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGImage.cpp:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGResource.cpp:
* Source/WebCore/style/StyleResolveForDocument.cpp:
* Source/WebCore/svg/SVGAElement.cpp:
* Source/WebCore/svg/SVGTextContentElement.cpp:
* Source/WebCore/testing/Internals.cpp:
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/mac/WKDOMRange.mm:
* Source/WebKit/WebProcess/InjectedBundle/DOM/InjectedBundleRangeHandle.cpp:
* Source/WebKit/WebProcess/WebPage/FindController.cpp:
* Source/WebKit/WebProcess/WebPage/WebFoundTextRangeController.cpp:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
* Source/WebKitLegacy/mac/DOM/DOM.mm:
* Source/WebKitLegacy/mac/DOM/DOMRange.mm:
* Source/WebKitLegacy/mac/DOM/WebDOMOperations.mm:
* Source/WebKitLegacy/mac/WebCoreSupport/WebValidationMessageClient.mm:
* Source/WebKitLegacy/mac/WebView/WebFullScreenController.mm:
* Tools/TestWebKitAPI/Tests/WebCore/MarkedText.cpp:

Canonical link: https://commits.webkit.org/294370@main
